### DISCUSSION TITLE
Shop - Purchase to Inventory

### DISCUSSION THREAD

**Discussion by Kaillera** [FakeFishGames/Barotrauma#8970] [Ideas and suggestions]
I feel like there are some things that the players feels like they must have now when shopping in outposts before they set off (especially wires and components). Would be great if they can just purchase it directly into their inventory instead of having to wait for the level to start. You don't want some random person swiping your hard earned salary money because they spawned closer to cargo or something.

**Comment by mLuby** [comment]
The problem with mid-round deliveries is players could acquire a huge number of _separate_ crates and deconstruct them into Steel Bars. A simple solution is to make crates not deconstruct into anything.

The ["hired hand" suggestion](https://github.com/Regalis11/Barotrauma/discussions/8935) could work nicely with this idea: when you exit the vendor interface, the items (in crates if >8 items) spawn by the vendor's feet. (Tangent: new hires should also spawn.) You can then carry the stuff to your sub yourself, or you can pay station dwellers to carry the stuff on the station's floor to your sub's cargo spawners (ignoring stolen stuff, of course). If the movers get stuck for a while, they just drop what they're carrying and leave.

Bonus: pay to have station dwellers carry all items (that they can access) _from_ your current sub _to_ the station floor and drop them. Then you can swap to a different sub and pay again to have all items on the station floor delivered to the sub's cargo spawners. Aside from being more convenient it'd add significant immersion. Also opens the door for events like "someone stole something during the move" and station reputation-related things like "stations will only help you move if rep > 50." (Might even try to steal from you or your sub if reputation < 50! Post guards.)

**Reply by Kaillera** [reply]
Actually they have changed that from the previous "3 steel bars from one metal crate" a long while back already. The steel itself isn't so much of a big deal.

**Reply by James-E-A** [reply]
> The problem with mid-round deliveries is players could acquire a huge number of separate crates and deconstruct them into Steel Bars. A simple solution is to make crates not deconstruct into anything.

If I'm reading the OP correctly, no crate would be involved: you _can't_ fit crates in your inventory, anyway—the idea would be just to (say) talk to the Engineering shopkeep, ask him for wires, and he'd _give them directly to the player_ (sort of like how Vending Machines work: purchases with no crates or round-delays involved)

**Reply by Kaillera** [reply]
Yeah James read it correctly

**Comment by James-E-A** [comment]
Improvement suggestion: _only_ allow this from department-specific shopkeeps

Say you buy a component from Engineering, or medicine from the Medical Officer, you'll have the option to just take it from the person directly—but if you ask the quartermaster (generic "Store" who sells everything) you'll have to wait 0.5 business days for shipping in a crate
